lk2nd: device: Add an LED driver (#512)
msm8916-mainline/lk2nd: Documentation/dt-bindings.md

		};
	};
```

### GPIO leds

> We also support the `gpio-leds` node. This node follows upstream binding except

Reviewer: ```suggestion
lk2nd supports the `gpio-leds` node. This node follows upstream binding except
```

My personal preference is to write documentation from the point of the software, not the people making it

that most properties are not supported. Currently only `gpios` and
`default-state` are supported in lk2nd.

```
gpio-leds {
    compatible = "gpio-leds";

    red {
        gpios = <&tlmm 3 GPIO_ACTIVE_HIGH>;
        default-state = "on";
    };

    green {
        gpios = <&tlmm 4 GPIO_ACTIVE_HIGH>;
        default-state = "keep";
    };

> 

Reviewer: > should we use "keep" or "on"/"off"? I prefer not using "keep" for clarity.

I feel like "keep" makes no sense for lk1st and such a simple driver, since the only goal for us is to set initial state of the led (most often to "on"). I suggest just omitting it here and in the driver (will just be a warning anyway)

Author: It is a preperty supported by  linux gpio-driver. I think it's better to be a no-op without a warning. It's different with some other random definitely wrong things. But l think it's also okay to omit it anyway.

    blue {
        gpios = <&tlmm 4 GPIO_ACTIVE_HIGH>;
        default-state = "off";
    };
};
```

This driver is mainly for lk1st since no previous boot-loader is capable to
light up an LED in this case. This makes it unclear what state currently the
device is in.  It's recommended to document all the LEDs here if lk1st is

> available for your device.

Reviewer: > "It's recommended to document"

I'd actually word this more softly, like "if you wish to make sure some leds are enabled or disabled by lk1st on boot, you can use this driver"

> Unlike linux, we don't want to introduce an LED subsystem. There is no public
> API for LED operations like turning LEDs on/off. And there is no LED trigger
> (yet). The only purpose for this driver is for debugging and the LED behavior
> consistancy between lk1st and lk2nd.

Reviewer: ```suggestion
lk2nd doesn't implement a fully-fledged led subsystem. The only purpose for
this driver is for debugging and the LED behavior consistancy between lk1st
and lk2nd.
```

Author: Seems reasonable.
